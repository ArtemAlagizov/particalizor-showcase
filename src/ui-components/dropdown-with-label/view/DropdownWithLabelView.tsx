import React from "react";
import { Styled } from "../styled";
import { IDropdownWithLabelView } from "./IDropdownWithLabelView";
import { Select } from "antd";

const { Option } = Select;

const DropdownWithLabelView: React.FC<IDropdownWithLabelView> = ({
  onChange,
  options,
  defaultValue,
  label
}) => {
  const dropdownOptions = options.map((option, index) => (
    <Option value={option.value} key={index}>
      {option.title}
    </Option>
  ));

  return (
    <Styled.DropdownWithLabelContainer>
      <Styled.DropdownLabel>{label}</Styled.DropdownLabel>
      <Styled.DropdownContainer>
        <Select
          defaultValue={defaultValue}
          style={{ width: "100%" }}
          onChange={onChange}
        >
          {dropdownOptions}
        </Select>
      </Styled.DropdownContainer>
    </Styled.DropdownWithLabelContainer>
  );
};

export default DropdownWithLabelView;
