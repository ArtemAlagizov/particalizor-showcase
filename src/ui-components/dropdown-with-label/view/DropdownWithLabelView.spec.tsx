import React from "react";
import { shallow } from "enzyme";
import DropdownWithLabelView from "./DropdownWithLabelView";

describe("Given a DropdownWithLabelView", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <DropdownWithLabelView
          options={[]}
          onChange={jest.fn()}
          label={""}
          defaultValue={""}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
