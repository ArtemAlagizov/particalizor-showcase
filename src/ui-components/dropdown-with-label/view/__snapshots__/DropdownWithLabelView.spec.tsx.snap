// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a DropdownWithLabelView when rendering should match snapshot 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <Select
      choiceTransitionName="zoom"
      defaultValue=""
      onChange={[MockFunction]}
      showSearch={false}
      style={
        Object {
          "width": "100%",
        }
      }
      transitionName="slide-up"
    />
  </styled.div>
</styled.div>
`;
